lib/pbench/server/api/resources/api_key.py

```diff
@@ -0,0 +1,104 @@
+from http import HTTPStatus
+
+from flask import jsonify, make_response
+from flask.wrappers import Request, Response
+from sqlalchemy.exc import IntegrityError, SQLAlchemyError
+
+from pbench.server import PbenchServerConfig
+from pbench.server.api.resources import (
+    APIAbort,
+    ApiAuthorizationType,
+    ApiBase,
+    ApiContext,
+    APIInternalError,
+    ApiMethod,
+    ApiParams,
+    ApiSchema,
+)
+import pbench.server.auth.auth as Auth
+from pbench.server.database.models.api_key import APIKeyError, APIKeys
+from pbench.server.database.models.audit import (
+    Audit,
+    AuditStatus,
+    AuditType,
+    OperationCode,
+)
+
+
+class APIKey(ApiBase):
```

Comment:
Technically, having both `pbench.server.database.models.APIKey` and `pbench.server.api.resources.APIKey` isn't a problem for Python or necessarily for us, and since we generally have only one reference outside the class to the API implementation name, the risk of confusion is small.

Maybe something like `APIKeyManage`? 🍤